codegen/src/api/mod.rs

```diff
@@ -461,14 +460,10 @@ impl RuntimeGenerator {
                 // Identify the pallets composing the static metadata by name.
                 pub static PALLETS: [&str; #pallet_names_len] = [ #(#pallet_names,)* ];
 
-                /// The statically generated runtime call type.
-                pub type Call = #runtime_path::RuntimeCall;
-
                 /// The error type returned when there is a runtime issue.
                 pub type DispatchError = #types_mod_ident::sp_runtime::DispatchError;
 
-                // Make the runtime event type easily accessible, and impl RootEvent to help decode into it.
-                pub type Event = #runtime_path::RuntimeEvent;
+                #outer_event
```

Comment:
Random thought while I remember; some pallets like sudo and multisig have calls that accept a `RuntimeCall` as a param. `RuntimeCall` is the outer enum for calls for every pallet, which means it prob points to a lot of types across pallets. I'd guess that this is what accounts for a bunch of extra size in those pallets.

Perhaps this will just improve when I substitute `RuntimeCall` for the Subxt call type (substituting a type is supposed to lead to that type not being generated any more in codegen, though I'm not 100% sure whether that works or not).

Reply:
I'm leaning towards allowing somebody to transcode a call into some other type eg RuntimeCall now actually; the problem with substituting RuntimeCall for some `BoxedPayload` is that the things using `RuntimeCall` require Encode,Decode,EncodeAsType,DecodeAsType,Debug at minimum, and there's not a sensible way to impl `Encode` and `Decode` for our `BoxedPayload`.

Transcoding is more "expensive", but can be opted out of if performance is important (just write the RuntimeCall enum manually), is less intrusive to the generated types (we don't need another substitution), and means we can avoid implementing non-functional Encode/Decode impls on our BoxedPayload.

Reply:
I just strip the RuntimeCall now that we've exposed these types in `scale-info`, so smaller metadata across the board :)